visualization/app/codeCharta/util/reduxHelper.ts

```diff
@@ -21,10 +21,10 @@ export function isActionOfType(actionType: string, actions) {
 	return actions[actionType] !== undefined
 }
 
-function arrayContainsItem(array: any[], item: any): boolean {
-	return array.some(x => isEqualObject(x, item))
+export function isEqualObject(obj1: any, obj2: any): boolean {
+	return angular.toJson(obj1) === angular.toJson(obj2)
```

Comment:
The reason to use `angular.toJson` is that angular adds hashKey-properties to objects which are using in the view. That hashkey is different every time.

We could write our own compare function which ignores this property, but this worked for perfectly for now.

Your argument is absolutely valid though. This can be improved, but I'm not sure if it is worth the time.

Reply:
Note: `JSON.stringify({ a: true, b: false }) !== JSON.stringify({ b: false, a: true })`

It is possible to use `track by`. That makes sure the objects in the array are tracked by their index in the array instead of a hash value. See e.g., https://stackoverflow.com/questions/18826320/what-is-the-hashkey-added-to-my-json-stringify-result

Reply:
Adressed in #1001